datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/DatalakeDrClient.java

```diff
@@ -256,12 +255,16 @@ public DatalakeBackupInfo getLastSuccessBackup(String datalakeName, String actor
             ListDatalakeBackupResponse response = newStub(channelWrapper.getChannel(), UUID.randomUUID().toString(), actorCrn)
                     .listDatalakeBackups(builder.build());
 
-            if (response != null && response.getDatalakeInfoList() != null) {
-                datalakeBackupInfo = response.getDatalakeInfoList().stream()
+            if (response != null) {
+                DatalakeBackupInfo datalakeBackupInfo = response.getDatalakeInfoList().stream()
                         .filter(backup -> "SUCCESSFUL".equals(backup.getOverallState()))
-                        .findFirst().orElse(null);
+                        .filter(backup -> runtime.isEmpty() || backup.getRuntimeVersion().equalsIgnoreCase(runtime.get()))
+                        .findFirst()
+                        .orElse(null);
+                LOGGER.info("The following successful backup was found for data lake {} and runtime {}: {}", datalakeName, runtime, datalakeBackupInfo);
```

Comment:
We'd still log this if there was no backups found.

Reply:
Yes and I'd consider that extra information worth logging.

Reply:
Ok, both logs are there now